pkg/reporter/reporter_test.go

```diff
@@ -255,11 +226,82 @@ var _ = PDescribe("Reporter", func() {
 
 		close(done)
 	}, 20)
+
+	Describe("AdditionalFields logic", func() {
+		When("AdditionalFields are defined on the meter def", func() {
+			It("Should add the additional fields found on the meterdefintion query to the meter report's additionalLabels", func(done Done) {
+
+				additionalFields := Keys{
+					"test_field_1": Equal("test-value-1"),
+					"test_field_2": Equal("test-value-2"),
+				}
+
+				for label, matcher := range additionalFields {
+					additionalLabels[label] = matcher
+				}
+
+				sut.meterDefinitions[0].Spec.Workloads[0].MetricLabels[0].Query = "rate(rpc_durations_seconds_count{test_field_1=test-value-1,test_field_2=test-value-2}[5m])*100"
```

Comment:
Preconditions should be set in a BeforeEach() - You can do that in the when before the It.